### ISSUE TITLE
Working 3D animation pipeline

### ISSUE THREAD

**Issue by gwald** [horde3d/Horde3D#239]
Hi,
Just wondering what 3D animation tools that work with the Horde3D 's converter.
There's openCollada with Max/Maya: 
http://horde3d.org/forums/viewtopic.php?f=11&t=2285

And I found TrueSpace working also: 
http://horde3d.org/forums/viewtopic.php?f=11&t=2285&p=10422&hilit=truespace#p10422
The problem with TS is that the import side is very limited, I'm testing these FBX, DAE and GLTF plugins: 
https://clintons3d.com/plugins/truespace/importexport/index.html

There is also skinned animation, aka vert animation/morphing, which is easier and works ,but only useful for a single frame of animation/tween:
https://github.com/horde3d/Horde3D/issues/225

Also the .anim format seems like a very simple format, I was thinking BVH animation would translate well to it, and probably a lot easier to get and more correct than Collada XML. 
What do you think of the convertor also using .bvh files for animation?

**Comment by algts** [comment]
Blender seems to be working correctly with version > 2.80 (it seems to be based on OpenCollada).
akeytsu can only export FBX. Don't know about other DCC software but they probably also use FBX.
Can't say anything about BVH animation, didn't work with it. Anyway, as engine only supports .anim files, you would have to convert the input format (collada, bvh, fbx, gltf, etc.) to anim file.
I think that extending converter is the way to go (should probably be called Asset converter when several formats are supported).

**Comment by gwald** [comment]
Good points, your right about akeytus, it imports collada but only exports FBX, the rest I know export collada.
I think I'm going to focus on just using this: FBX 2013.3 Converter for Windows 64-bit
 https://aps.autodesk.com/developer/overview/fbx-converter-archives

It had this issue: https://github.com/horde3d/Horde3D/pull/240
And seems to have created the model and anim, but I'm getting...

Invalid node handle in h3dSetupModelAnimStage	
Invalid node handle in h3dSetModelAnimParams
and
Invalid node handle in h3dUpdateModel

**Comment by gwald** [comment]
I was trying to animate a group node :/
https://github.com/horde3d/Horde3D/issues/241

FBX 2013.3 Converter seems to be pretty good so far.

I'll leave this open and test a few of the 3D apps.








---------------------

Blender exports works but you have to export with only the basics, ie triangles, normals, UVs, tangents, and skeleton and animations.
Everything else needs to be baked in or disable on export.

I would recommend getting another 3D application for .dae validation, I use  [https://www.assimp.org/](https://www.assimp.org/) for test/converting.
But there's also MeshLab, Wings 3D, SketchUp FreeCAD etc.
Basically, if they have big problems with your .dae then there's no point continuing to ColladaConv.

I recommend having ColladaConv debugable in your development environment so you can see what it's doing.
Or using the [develop branch](https://github.com/horde3d/Horde3D/blob/develop/Horde3D/Source/ColladaConverter/main.cpp#L140) which has a -verbose option which is useful to see where it get's stuck for clues.

And don't forget that you have to [create the texture and shader XML files manually](https://github.com/horde3d/Horde3D/issues/219).
